CsvUtil.cs

```diff
@@ -216,41 +218,55 @@ public static class CsvUtil {
             int n = 0;
             foreach(string field in EnumerateCsvLine(header)) {
                 var trimmed = field.Trim();
-                if (!trimmed.StartsWith("#"))
+                if (!trimmed.StartsWith("#")) {
+                    trimmed = RemoveSpaces(trimmed);
                     headers[trimmed] = n;
+                }
                 ++n;
             }
             return headers;
         }
 
         // Parse an object line based on the header, return true if any fields matched
-        private static bool ParseLineToObject(string line, Dictionary<string, int> fieldDefs, FieldInfo[] fi, object destObject, bool strict) {
+        private static bool ParseLineToObject(string line, Dictionary<string, int> fieldDefs, FieldInfo[] fi, PropertyInfo[] pi, object destObject, bool strict) {
 
             string[] values = EnumerateCsvLine(line).ToArray();
             bool setAny = false;
             foreach(string field in fieldDefs.Keys) {
                 int index = fieldDefs[field];
                 if (index < values.Length) {
                     string val = values[index];
-                    setAny = SetField(field, val, fi, destObject) || setAny;
+                    setAny = SetField(field, val, fi, pi, destObject) || setAny;
                 } else if (strict) {
                     Debug.LogWarning(string.Format("CsvUtil: error parsing line '{0}': not enough fields", line));
                 }
             }
             return setAny;
         }
 
-        private static bool SetField(string fieldName, string val, FieldInfo[] fi, object destObject) {
+        private static bool SetField(string fieldName, string val, FieldInfo[] fi, PropertyInfo[] pi, object destObject) {
+            bool result = false;
+            foreach (PropertyInfo p in pi) {
+                // Case insensitive comparison
+                if (string.Compare(fieldName, p.Name, true) == 0) {
+                    // Might need to parse the string into the property type
+                    object typedVal = p.PropertyType == typeof(string) ? val : ParseString(val, p.PropertyType);
+                    p.SetValue(destObject, typedVal, null);
+                    result = true;
+                    break;
+                }
+            }
             foreach(FieldInfo f in fi) {
```

Comment:
I think I would skip iterating the fields if `result == true`, to avoid double-setting a value if its property and private field are called the same thing (with case difference). It probably wouldn't do any harm since the property set is first but I imagine people would expect properties to be preferred?

Reply:
Ah nevermind, I came to make this change myself and saw in the tests that actually there are cases where potentially you could need to set both - probably a bad class setup but even so, could deviate. Will merge as-is.